pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/tests/Test_LightningKokkosMeasures.cpp

```diff
@@ -1777,3 +1777,46 @@ TEST_CASE("Measurement with a seeded device", "[Measures]") {
         CHECK(*m == *m1);
     }
 }
+
+TEST_CASE("Sample with a seeded device", "[Measures]") {
+    for (std::size_t _ = 0; _ < 5; _++) {
+        std::unique_ptr<LKSimulator> sim = std::make_unique<LKSimulator>();
+        std::unique_ptr<LKSimulator> sim1 = std::make_unique<LKSimulator>();
+
+        std::size_t shots = 100;
+
+        std::vector<double> samples(shots * 4);
+        MemRefT<double, 2> buffer{
+            samples.data(), samples.data(), 0, {shots, 1}, {1, 1}};
+        DataView<double, 2> view(buffer.data_aligned, buffer.offset,
+                                 buffer.sizes, buffer.strides);
+
+        std::vector<double> samples1(shots * 4);
+        MemRefT<double, 2> buffer1{
+            samples1.data(), samples1.data(), 0, {shots, 1}, {1, 1}};
+        DataView<double, 2> view1(buffer1.data_aligned, buffer1.offset,
+                                  buffer1.sizes, buffer1.strides);
+
+        std::mt19937 gen(37);
```

Comment:
Since we can generate 2 RNGs and reuse them here, we can also put these outside the loop.